### PR TITLE
Prevent duplicate jobs from running, resolves mheily/relaunchd#9

### PR DESCRIPTION
**Pull request by sgerbino**
This patch shoud resolve issue #9.

The code change produces the following output:

``` bash
[sgerbino@elysium relaunchd]$ ./launchd 
[sgerbino@elysium relaunchd]$ sh launchctl load test/fixtures/sleepy.json 
[sgerbino@elysium relaunchd]$ sh launchctl load test/fixtures/sleepy.json 
[sgerbino@elysium relaunchd]$ sh launchctl load test/fixtures/sleepy.json 
[sgerbino@elysium relaunchd]$ sh launchctl load test/fixtures/sleepy.json 
[sgerbino@elysium relaunchd]$ sh launchctl load test/fixtures/sleepy.json 
[sgerbino@elysium relaunchd]$ sh launchctl list
PID      Status   Label
29394    0        sleepy
[sgerbino@elysium relaunchd]$ cat ~/.launchd/launchd.log 
update_jobs(manager.c:172): WARNING: job sleepy is already running
update_jobs(manager.c:172): WARNING: job sleepy is already running
update_jobs(manager.c:172): WARNING: job sleepy is already running
update_jobs(manager.c:172): WARNING: job sleepy is already running
```

### CODE REVIEW

File: manager.c [hunk 1]
```diff
@@ -160,11 +160,35 @@ void update_jobs(void)
...
+				log_warning("job `%s' is already running", jm->label);
+				skip = true;
+				break;
+			}
+		}
+
+		if (skip)
+			continue;
+		
+		LIST_FOREACH(job, &jobs, joblist_entry) {
```
Comment (mheily) [inline manager.c]: From here down, it looks like duplicate code with a slightly different warning message.

**Comment by mheily** [comment]
I created a utility function for looking up jobs by label, and committed it in  3283b77. Let's refactor this patch to use the utility function.

File: manager.c (hunk 1 above)
Comment (sgerbino) [inline manager.c]: Oops, you're right, the second loop should be iterating around joblist instead of jobs. I'll fix that up and use your utility function.

**Comment by sgerbino** [comment]
The first loop should prevent jobs from running that are already running.
The second loop is to prevent loading the same job twice if 2 of the same job are in the pending LIST.

**Pull request closed without merging**